modules/perception/camera/tools/offline/offline_obstacle_pipeline.cc

```diff
@@ -259,18 +293,54 @@ int work() {
     perception.GetCalibrationService(&frame.calibration_service);
 
     // save distortion images
-    std::string save_dir = FLAGS_undistortion_save_dir + "/" + camera_name;
+    std::string save_dir = FLAGS_undistortion_save_dir;
     if (FLAGS_do_undistortion && (FLAGS_undistortion_save_dir != "") &&
         cyber::common::PathExists(save_dir)) {
       base::Image8U image1;
       DataProvider::ImageOptions image_options;
       image_options.target_color = base::Color::BGR;
       frame.data_provider->GetImage(image_options, &image1);
       save_image(save_dir + "/" + image_name + FLAGS_image_ext, image1);
+      AINFO << "Undistorted image saved to : "
+            << save_dir + "/" + image_name + FLAGS_image_ext;
     }
 
     CHECK(perception.Perception(options, &frame));
     visualize.ShowResult(image, frame);
+
+    save_dir = FLAGS_save_dir;
+    if (!cyber::common::PathExists(save_dir)) {
+      AERROR << "save_dir does not exist : " << save_dir;
+    } else {
+      std::ofstream myfile;
+      myfile.open(save_dir + "/" + image_name + ".txt");
+      for (auto obj : frame.detected_objects) {
+        auto &box = obj->camera_supplement.box;
+        myfile << base::kSubType2NameMap.at(obj->sub_type) + " " +
+                  std::to_string(static_cast<int>(box.xmin)) + " " +
```

Comment:
No need to call "+" and "std::to_string()", as the ofstream will handle them correctly.